docs/openapi.json

```diff
@@ -380,10 +380,30 @@
                     "slurmcern"
                   ]
                 },
+                "default_kubernetes_jobs_timeout": {
+                  "title": "Default timeout for Kubernetes jobs",
+                  "value": "604800"
+                },
                 "default_workspace": {
                   "title": "Default workspace",
                   "value": "/usr/share"
                 },
+                "kubernetes_max_memory_limit": {
+                  "title": "Maximum allowed memory limit for Kubernetes jobs",
+                  "value": "10Gi"
+                },
+                "kubernetes_memory_limit": {
+                  "title": "Default memory limit for Kubernetes jobs",
+                  "value": "3Gi"
+                },
+                "maximum_kubernetes_jobs_timeout": {
+                  "title": "Maximum timeout for Kubernetes jobs",
+                  "value": "1209600"
+                },
+                "maximum_workspace_retention_period": {
+                  "title": "Maximum retention period in days for workspace files",
+                  "value": "3650"
```

Comment:
BTW the default REANA installation leads to:

```console
$ reana-client info | grep None
Maximum allowed memory limit for Kubernetes jobs: None
Maximum retention period in days for workspace files: None
```

None may be perhaps leading to misinterpretations, but dunno whether we can do something better. Perhaps hide null values in the client? Just thinking out loud...

```console
$ reana-client info --json | jq
{
  "kubernetes_max_memory_limit": {
    "title": "Maximum allowed memory limit for Kubernetes jobs",
    "value": null
  },
  "maximum_kubernetes_jobs_timeout": {
    "title": "Maximum timeout for Kubernetes jobs",
    "value": "1209600"
  },
  "maximum_workspace_retention_period": {
    "title": "Maximum retention period in days for workspace files",
    "value": null
  }
}

```

Reply:
I can detect if it is `None` and replace it with a string like `Unlimited maximum retention period` or `Keep forever`.

Example:

```json
 "maximum_workspace_retention_period": {
    "title": "Maximum retention period in days for workspace files",
    "value": "No limit. Keep forever",
  }
```

WDYT?

Reply:
Do you mean to keep `null` in JSON and replace `None` with a string such as "Not set" on the client-side only for the textual output? Or do you mean to massage the output already on the server-side so that a string such as "Not set" is returned by the server REST API side already? 

(BTW if we want to better expose default values to the user, instead of just saying "not set" for N endpoints and leaving the user to wonder how to detect default value, we could indeed generate nicer phrases such as "Keep forever" on the server-side, although they won't be very configurable then.  Alternatively, we could transform these special strings such as "forever" to special values such as 9999999999999 to represent infinity; in this way no unconfigurable strings would be necessary, but it would would be a bigger change that would go against "optionality" 
 of values and types a bit.)

Reply:
I meant to change it on the server side. But it might as well work on the client side. Maybe, even better, cause if the client receives `null/None`, you can act on it knowing no limit exists and later display it to the user like `No limit`.

Reply:
So we will keep JSON output with `null/None` and user response with a user-friendly message.

```console
> reana-client info
List of supported compute backends: kubernetes
Default timeout for Kubernetes jobs: 604800
Default memory limit for Kubernetes jobs: 4Gi
Default workspace: /var/reana
Maximum allowed memory limit for Kubernetes jobs: no limit
Maximum timeout for Kubernetes jobs: 1209600
Maximum retention period in days for workspace files: no limit
List of available workspaces: /var/reana
```

And if `--json` passed, it will return this:

```json
{"compute_backends": {"title": "List of supported compute backends", "value": ["kubernetes"]}, "default_kubernetes_jobs_timeout": {"title": "Default timeout for Kubernetes jobs", "value": "604800"}, "default_kubernetes_memory_limit": {"title": "Default memory limit for Kubernetes jobs", "value": "4Gi"}, "default_workspace": {"title": "Default workspace", "value": "/var/reana"}, "kubernetes_max_memory_limit": {"title": "Maximum allowed memory limit for Kubernetes jobs", "value": null}, "maximum_kubernetes_jobs_timeout": {"title": "Maximum timeout for Kubernetes jobs", "value": "1209600"}, "maximum_workspace_retention_period": {"title": "Maximum retention period in days for workspace files", "value": null}, "workspaces_available": {"title": "List of available workspaces", "value": ["/var/reana"]}}
```

Reply:
@mdonadoni mentioned that it might be nice to group values together for better output ([link](https://github.com/reanahub/reana-server/pull/540#pullrequestreview-1069850777)). I can create an issue to improve this and, in addition, figure out this `null/None` and user-friendly message.

Reply:
WRT grouping, that we can leave for later. It should be enough if the values are sorted alphabetically and if similar symbols and near each other.  This is why I usually prefer common prefix rather than common suffix, e.g. `kubernetes_memory_limit_max` and `..._min`  vs ` max_kubernetes_memory_limit` and `min_...`.  In this way, the grouping is occurring naturally thanks to alphabetical ordering. (But it may not be practical to change the naming of all these symbols now if they are already exposed elsewhere...)